src/Microsoft.DotNet.Wpf/specs/tooltip.md

```diff
@@ -0,0 +1,305 @@
+
+The document includes the following sections:
+- **Requirements**.  Links to sources of information that drove the decisions described here.
+- **Applicability**.  How the new behavior will be delivered.
+- **Conceptual pages**.  Specifies tooltip behavior.  The emphasis is on new behavior, but old behavior is included when helpful for context.  This is the public-facing spec, as it might appear in "overview" or "how-to" pages on docs.microsoft.com. 
+- **API Pages**.  Public-facing spec for new APIs, as it might appear in "reference" pages on docs.microsoft.com.
+- **Implementation**.  Internal-facing spec - details needed by implementers, testers, and reviewers, but not part of the public documentation.
+- **Discussion**.  Remarks that aren't part of the spec, but help understand it.  Including how the spec meets the requirements, goals and non-goals, alternatives that were considered, etc.  
+- **Summary of behavior changes**.  Consolidated in one place for convenience.
+- **API Details**.  Consolidated in one place for convenience.
+
+
+# Requirements 
+This spec cites the following sources (the [brackets] contain abbreviations used elsewhere in this document):
+
+1. WCAG 2.1, section 1.4.13: [Content on hover or focus](https://www.w3.org/TR/WCAG21/#content-on-hover-or-focus) [WCAG] 
+2. WCAG 2.1:  [Understanding 1.4.13](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html)   [WCAG]
+3. MAS 1.4.13:  [Content on hover or focus](https://microsoft.sharepoint.com/:w:/r/sites/accessibility/_layouts/15/Doc.aspx?sourcedoc=%7B56729308-859e-4b32-ab27-04d4364b3003%7D&action=view&wdLOR=c9C9640FB-F04E-401A-BF46-A79887F322D3&wdAccPdf=0&wdparaid=38854747) [MAS]
+4. The [Figma diagram](https://www.figma.com/file/8LeT0Eq7a3ylAyM19c0FJp/ASC%3A-Tooltip?node-id=110%3A0) (as of February 2021) [FD]
+5. Email thread (MS internal) [ET] 
+6. WPF [Tooltip Overview](https://docs.microsoft.com/en-us/dotnet/desktop/wpf/controls/tooltip-overview?view=netframeworkdesktop-4.8) (and related links) [TO]
+
+The first three describe behavior in general terms, guiding any UI platform (WCAG) or Microsoft UI platforms (MAS).
+The Figma diagram has more specific requirements, but is not a complete spec.
+The email thread among many relevant MS teams helped fill in omissions, corrected errors, and clarified ambiguities in the diagram.
+The Tooltip Overview documents the existing behavior in WPF as of .NET 4.7.2.
+
+# Applicability
+
+All the changes described here are required to appear in .NET 6.0.
+
+The behavior changes will also appear in servicing updates for earlier versions as follows:
+- .NET 4.6-4.7.2, .NET 4.8:  the next biannual accessibility update (expected summer 2021)
+- .NET Core 3.1, .NET 5.0:  same time frame as above, patch version TBD
+
+The API changes (and behavior that depends on them) are only in .NET 6.0;
+servicing updates cannot include changes to public API.
+
+In the servicing updates, the new behavior is opt-in by setting
```

Comment:
If I understood correctly, it's supposed to be use the new behavior by default but developers can opt-out of the new behavior by setting `Switch.UseLegacyAccessibilityFeatures.5` to false.

I would suggest changing it, unless I misunderstood the spec.
```suggestion
In the servicing updates, the new behavior is enabled by default but developers can opt-out by setting
```

Reply:
It's correct as written.  Setting the switch to true means "use the old behavior".   Runtimes that know about the switch (i.e. 6.0 and patched versions of older runtimes) use the target version of app to set the default value for the switch, in this case defaulting to false if the app targets 6.0 and true otherwise.  Thus apps that target 6.0 get the new behavior, apps that target older versions get the old behavior.  An older app that wants the new behavior has to opt-in by setting the switch to false.  (More literally, the app opts out of using the old behavior.)

This confuses everyone the first time you see it.  But eventually you realize that it's the right design (it took me a few weeks).   Your assertion
>it's supposed to be use the new behavior by default 

isn't quite right.  It's true for apps that target the new runtime, but older apps should use the old behavior by default (for compat).

Reply:
You're right, I read the Applicability section again and you're comment and I understand what you mean. It's for apps targeting older frameworks than .Net 6 to be able to have the accessibility behavior of .Net 6. So, apps that target .Net 6 uses the new behavior by default but developers can opt-out and apps that target the frameworks getting a servicing update can opt-in.

Thank you!